### PR TITLE
Add prodigal (new attempt)

### PR DESCRIPTION
**Pull request by Darcy220606**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [] If you've fixed a bug or added code that should be tested, add tests!
  - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by Darcy220606** [comment]
BUT NOTE: When the pipeline is run locally with --run_annotaion_tool prodigal i still get the error
`WARN: There's no process matching config selector: PRODIGAL -- Did you mean: PROKKA?`

File: docs/output.md [hunk 1]
```diff
@@ -60,7 +61,7 @@ Antimicrobial Resistance Genes (ARGs):
 Antimicrobial Peptides (AMPs) and peptide annotation:
 
 - [Prokka](#prokka) - (default) open reading frame and functional protein annotation
-  <!--* [prodigal](#prodigal) - (optional: replaces prokka) open reading frame annotation-->
+- [Prodigal](#prodigal) - (optional: replaces prokka) for open reading frame annotation
```
Comment (jfy133) [inline docs/output.md:64]:
```suggestion
- [Prodigal](#prodigal) - for open reading frame annotation
```

File: docs/output.md [hunk 2]
```diff
@@ -60,7 +61,7 @@ Antimicrobial Resistance Genes (ARGs):
 Antimicrobial Peptides (AMPs) and peptide annotation:
 
 - [Prokka](#prokka) - (default) open reading frame and functional protein annotation
```
Comment (jfy133) [inline docs/output.md:63]:
```suggestion
- [Prokka](#prokka) - (optional: alternative to prodigal) open reading frame and functional protein annotation
```

File: nextflow.config [hunk 3]
```diff
@@ -18,6 +18,12 @@ params {
     igenomes_base              = 's3://ngi-igenomes/igenomes'
     igenomes_ignore            = true
 
+    // Annotation options
+    run_annotation_tool             = 'prodigal'
+
+    // Prodigal Annotation
+    my_prodigal_f           = 'gff'
```
Comment (jfy133) [inline nextflow.config:25]: `my_prodigal_f` is not very informative...

`annotation_prodigal_<whatever it refers to?>` might be better

File: workflows/funcscan.nf [hunk 4]
```diff
@@ -95,11 +95,21 @@ workflow FUNCSCAN {
                         .mix(fasta_prep.uncompressed)
 
     // Some tools require annotated FASTAs
-    // TODO only execute when we run tools that require prokka as input, e.g.
+    // TODO only execute when we run tools that require prokka/prodigal as input, e.g.
     // if ( params.run_bgc_tool1 | params.run_bgc_tool2 | params.run_bgc_tool3 ) etc.
```
Comment (jfy133) [inline workflows/funcscan.nf:99]:
```suggestion
```

I guess we don't need thi sanymore

File: workflows/funcscan.nf [hunk 5]
```diff
@@ -95,11 +95,21 @@ workflow FUNCSCAN {
...
+            PRODIGAL ( ch_prepped_input, params.my_prodigal_f )
+            ch_versions = ch_versions.mix(PRODIGAL.out.versions)
+            ch_annotation_output = PRODIGAL.out.amino_acid_fasta
+        }   else if ( params.run_annotation_tool == "prokka") {
+            PROKKA ( ch_prepped_input, [], [] )
+            ch_versions = ch_versions.mix(PROKKA.out.versions)
+            ch_annotation_output = PROKKA.out.faa
+        }
+    else {
+    ( ch_annotation_output = Channel.empty() )
```
Comment (jfy133) [inline workflows/funcscan.nf:111]:
```suggestion
        ( ch_annotation_output = Channel.empty() )
```

**Review by jfy133 — CHANGES_REQUESTED** [review]
A few tweaks, main one os the prodigal specific param name (I don't know what it does from the flag)

File: nextflow.config (hunk 3 above)
Comment (Darcy220606) [reply nextflow.config]: ok i just changed it to prodigal_output_format

**Comment by Darcy220606** [comment]
It seems there is a problem with `Deeparg_download_data` module. I ran a test locally (using conda) by skipping the arg_screening workflow and it all worked fine with prodigal. Any idea what might be the issue with deeparg?

**Comment by jfy133** [comment]
@Darcy220606 looking at the CI test logs:

```
 INFO:root:Downloading data into db/ Directory
  INFO:root:Downloading Models
  Traceback (most recent call last):
    File "/usr/local/bin/deeparg", line 10, in <module>
      sys.exit(main())
    File "/usr/local/lib/python2.7/site-packages/deeparg/entry.py", line 184, in main
      args.func(args)
    File "/usr/local/lib/python2.7/site-packages/deeparg/entry.py", line 82, in download_data
      download_file( "{}/model/{}/metadata_LS.pkl".format(main_url, model_v), "{}/model/{}/metadata_LS.pkl".format(args.output_path, model_v), "wb")
    File "/usr/local/lib/python2.7/site-packages/deeparg/entry.py", line 66, in download_file
      r = requests.get(url, stream = True)
    File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 76, in get
      return request('get', url, params=params, **kwargs)
    File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 61, in request
      return session.request(method=method, url=url, **kwargs)
    File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 542, in request
      resp = self.send(prep, **send_kwargs)
    File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 655, in send
      r = adapter.send(request, **kwargs)
    File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 498, in send
      raise ConnectionError(err, request=request)
  requests.exceptions.ConnectionError: ('Connection aborted.', error(104, 'Connection reset by peer'))

```

The database download server killed the connection for some reason. Just have to try with re-trying

File: docs/output.md [hunk 6]
```diff
@@ -59,8 +60,8 @@ Antimicrobial Resistance Genes (ARGs):
 
 Antimicrobial Peptides (AMPs) and peptide annotation:
 
-- [Prokka](#prokka) - (default) open reading frame and functional protein annotation
-  <!--* [prodigal](#prodigal) - (optional: replaces prokka) open reading frame annotation-->
+- [Prokka](#prokka) - (optional: alternative to prodigal) open reading frame and functional protein annotation
+- [Prodigal](#prodigal) - for open reading frame annotation
```
Comment (jfy133) [inline docs/output.md:64]: Makes sense this goes before Prokka, if prodigal is the alternative

File: docs/output.md [hunk 7]
```diff
@@ -35,6 +35,7 @@ work/
 ```bash
 outdir/
 ├── antismash/
+├── prodigal/
```
Comment (jfy133) [inline docs/output.md:38]: Please put in alphabetiocal order like the rest

File: nextflow_schema.json [hunk 8]
```diff
@@ -11,6 +11,11 @@
             "description": "",
             "default": "",
             "properties": {
+                "run_annotation_tool": {
+                    "type": "string",
+                    "default": "prodigal",
+                    "description": "Assign which annotation tool to run either prodigal or prokka"
```
Comment (jfy133) [inline nextflow_schema.json:17]: Should add an enum here, but I guess we cand o that later when we review the docs

File: workflows/funcscan.nf [hunk 9]
```diff
@@ -95,11 +95,19 @@ workflow FUNCSCAN {
...
+            ch_versions = ch_versions.mix(PRODIGAL.out.versions)
+            ch_annotation_output = PRODIGAL.out.amino_acid_fasta
+        }   else if ( params.run_annotation_tool == "prokka") {
+            PROKKA ( ch_prepped_input, [], [] )
+            ch_versions = ch_versions.mix(PROKKA.out.versions)
+            ch_annotation_output = PROKKA.out.faa
+        }
+    else {
+        ( ch_annotation_output = Channel.empty() )
+    }
```
Comment (jfy133) [inline workflows/funcscan.nf:110]:
```suggestion
        }
```

**Comment by jfy133** [comment]
Ok basically the FTP server from Virgina Tech is dead :/. I'm not sure what to there. IF you wanna try contacting the authors could see what happenss, oterhwise we have to wait a bit. But if it never comes back I don't know if there is any other workaround :/

File: nextflow_schema.json (hunk 8 above)
Comment (Darcy220606) [reply nextflow_schema.json:17]: sounds good

**Pull request merged**